packages/did/src/Did.signature.ts

```diff
@@ -59,23 +62,46 @@ function verifyDidSignatureDataStructure(
  * @param input.message The message that was signed.
  * @param input.signature Signature bytes.
  * @param input.keyUri DID URI of the key used for signing.
+ * @param input.expectedSigner If given, verification fails if the controller of the signing key is not the expectedSigner.
+ * @param input.allowUpgraded If `expectedSigner` is a light DID, setting this flag to `true` will accept signatures by the corresponding full DID.
  * @param input.expectedVerificationMethod Which relationship to the signer DID the key must have.
  * @param input.didResolveKey Allows specifying a custom DID key resolve. Defaults to the built-in [[resolveKey]].
  */
 export async function verifyDidSignature({
   message,
   signature,
   keyUri,
+  expectedSigner,
+  allowUpgraded = false,
   expectedVerificationMethod,
   didResolveKey = resolveKey,
 }: DidSignatureVerificationInput): Promise<void> {
   // Verification fails if the signature key URI is not valid
-  const { fragment } = parse(keyUri)
-  if (!fragment)
+  const signer = parse(keyUri)
+  if (!signer.fragment)
```

Comment:
Not introduced in this PR, but: doesn’t the didResolveKey below perform the same check? It could be removed from here then…

Reply:
it does, I was assuming this was introduced to provide better error reporting?

Reply:
Could it be that it just was not removed when the previous duplicated code was replaced with resolveKey?

Reply:
possibly. also the error on resolveKey is misleading... I'll fix that

Reply:
Nice!